src/Youtube.js

```diff
@@ -1,634 +1,672 @@
+Youtube.prototype.options_ = {};
 
-  Youtube.parseUrl = function(url) {
-    var result = {
-      videoId: null
-    };
+Youtube.apiReadyQueue = [];
 
-    var regex = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|\&v=)([^#\&\?]*).*/;
-    var match = url.match(regex);
+/* Youtube Support Testing -------------------------------------------------------- */
 
-    if (match && match[2].length === 11) {
-      result.videoId = match[2];
-    }
+Youtube.isSupported = function () {
+  return true;
+};
 
-    var regPlaylist = /[?&]list=([^#\&\?]+)/;
-    match = url.match(regPlaylist);
+// Add Source Handler pattern functions to this tech
+Tech.withSourceHandlers(Youtube);
 
-    if(match && match[1]) {
-      result.listId = match[1];
-    }
+/*
+ * The default native source handler.
+ * This simply passes the source to the video element. Nothing fancy.
+ *
+ * @param  {Object} source   The source object
+ * @param  {Flash} tech  The instance of the Flash tech
+ */
+Youtube.nativeSourceHandler = {};
+
+let _isOnMobile = /(iPad|iPhone|iPod|Android)/g.test(navigator.userAgent);
 
-    return result;
+Youtube.parseUrl = function (url) {
+  let result = {
+    videoId: null
   };
 
-  function loadApi() {
-    var tag = document.createElement('script');
-    tag.src = 'https://www.youtube.com/iframe_api';
-    var firstScriptTag = document.getElementsByTagName('script')[0];
-    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+  const regex = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|\&v=)([^#\&\?]*).*/;
+  let match = url.match(regex);
+
+  if (match && match[2].length === 11) {
+    result.videoId = match[2];
   }
 
-  function injectCss() {
-    var css = // iframe blocker to catch mouse events
-              '.vjs-youtube .vjs-iframe-blocker { display: none; }' +
-              '.vjs-youtube.vjs-user-inactive .vjs-iframe-blocker { display: block; }' +
-              '.vjs-youtube .vjs-poster { background-size: cover; }' +
-              '.vjs-youtube-mobile .vjs-big-play-button { display: none; }';
+  const regPlaylist = /[?&]list=([^#\&\?]+)/;
+  match = url.match(regPlaylist);
 
-    var head = document.head || document.getElementsByTagName('head')[0];
+  if (match && match[1]) {
+    result.listId = match[1];
+  }
 
-    var style = document.createElement('style');
-    style.type = 'text/css';
+  return result;
+};
 
-    if (style.styleSheet){
-      style.styleSheet.cssText = css;
-    } else {
-      style.appendChild(document.createTextNode(css));
-    }
+const loadApi = function () {
+  let tag = document.createElement('script');
+  tag.src = 'https://www.youtube.com/iframe_api';
+  let firstScriptTag = document.getElementsByTagName('script')[0];
+  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+};
+
+const injectCss = function () {
+  let css = // iframe blocker to catch mouse events
+    '.vjs-youtube .vjs-iframe-blocker { display: none; }' +
```

Comment:
template strings can be used here instead of string concatenation. Will make this block a bit nicer.